src/mongo/client/dbclientinterface.h

```diff
@@ -1080,7 +1081,7 @@ namespace mongo {
 
            See testDbEval() in dbclient.cpp for an example of usage.
         */
-        bool eval(const std::string &dbname, const std::string &jscode, BSONObj& info, BSONElement& retValue, BSONObj *args = 0);
+        bool eval(const std::string &dbname, const std::string &jscode, bool nolock, BSONObj& info, BSONElement& retValue, BSONObj *args = 0);
```

Comment:
I think default parameter is the most sane of those options.
To be honest, few users will likely even call this version of eval rather than one of the more sane overloads that I left unchanged.
I think introducing an `evalNolock` function would be overkill and would (further) clutter the API.